include/zephyr/drivers/video.h

```diff
@@ -202,6 +202,33 @@ struct video_frmival_enum {
 	};
 };
 
+/**
+ * @brief Port / Endpoint DT Helpers
+ *
+ */
+
+/* Drivers should not use this except they know exactly that port has a pid */
+#define _DT_INST_PORT_BY_ID(n, pid)                                                                \
+	COND_CODE_1(DT_NODE_EXISTS(DT_INST_CHILD(n, ports)), (DT_CHILD(DT_INST_CHILD(n, ports), port_##pid)), (DT_INST_CHILD(n, port_##pid)))
+
+/* Drivers rarely use this as they usually deals with only endpoint */
+#define DT_INST_PORT_BY_ID(n, pid)                                                                 \
+	COND_CODE_1(DT_NODE_EXISTS(_DT_INST_PORT_BY_ID(n, pid)), (_DT_INST_PORT_BY_ID(n, pid)), (DT_INST_CHILD(n, port)))
```

Comment:
I think it is possible to combine the two previous function in just one, as it is not permitted to have multiple ports (i.e. `port@0`, `port@1`, `port@2`...) without a `ports { ... };` node.

```suggestion
#define DT_INST_PORT_BY_ID(node, pid)                                   \
        COND_CODE_1(DT_NODE_HAS_CHILD(DT_INST_CHILD(node, ports)),      \
                    (DT_CHILD(DT_INST_CHILD(node, ports), port_##pid)), \
                    (DT_INST_CHILD(node, port)))

```